contrib/packaging_python/Readme.md

```diff
@@ -0,0 +1,123 @@
+# Packaging for PyPi
+
+Generally, PyPi packages with binary components are packaged as
+[wheels](https://packaging.python.org/en/latest/glossary/#term-Wheel).
+
+
+## Linux wheels
+
+Python wheels with C/C++ extensions need to be built on an old version of Linux
+so that they are supported on many platforms. The standard approach is to use
+the [manylinux](https://github.com/pypa/manylinux) Docker images as proposed in
+[PEP-513](https://www.python.org/dev/peps/pep-0513/).
+
+Most of the process for building wheels is automated by scripts in
+`contrib/packaging_python/` and is usually done for releases automatically by
+our CI.
+**PyPi does not allow reuploading any wheels with the same version number.**
+This means there is
+only one shot per version number (and per Python version), so it should work and
+be a release version.
+
+To build the python wheel from you current build directory, simply run
+`mk_wheel.py bdist_wheel -d dist` from the build directory.
+
+### Building wheels with manylinux2014
+
+The main script for building and uploading wheels is implemented in the
+`package-python-wheel` action. It builds a Docker container that is a slightly
+extended version of `manylinux2014` (see `manylinux2014/Dockerfile`), runs the `mk_clean_wheel.sh` script for every supported python version and finally collects and uploads the wheels.
+
+The `mk_clean_wheel.sh`:
+
+1. prepares the environment by creating and activating a proper python venv and installing some packages;
+2. configures cvc5 appropriately using the `mk_build_dir.py` script, which makes sure that the venv python version is used;
+3. builds cvc5 and the python extension via the `mk_wheel.py` script;
+4. postprocesses the wheel (with `auditwheel`) and moves it out of the build folder.
+
+
+## Uploading to PyPi
+
+To upload a wheel to test PyPi,
+
+    twine upload --repository testpypi -u $USERNAME -p $PASSWORD PATH_TO_WHEEL
+
+Note that you will need a TestPyPi login. Once it has been uploaded, you can
+test (from anywhere, not just the container) that the wheel works by installing
+it from the TestPyPi repository.
+
+```
+python3 -m pip install --index-url https://test.pypi.org/simple/ pycvc5==<cvc5 version number>
+python3 -c "import pycvc5; solver = pycvc5.Solver(); print(solver.getIntegerSort())"
+```
+
+You can remove the repository argument from the upload command to upload to
+real PyPi.
+
+## What goes on within the Docker container
+
+### Setup
+
+Compared to `manylinux2014`, the container comes with `ccache` and some cvc5
+dependencies preinstalled. Additionally, we point `cmake` to `ccache` using the `CMAKE_<LANG>_COMPILER_LAUNCHER` environment variables.
```

Comment:
> Additionally, we point `cmake` to `ccache` using the `CMAKE_<LANG>_COMPILER_LAUNCHER` environment variables.

It looks like that is not accurate anymore. Instead, the container creates symlinks.